packages/realm/scripts/submit-analytics.mjs

```diff
@@ -16,6 +16,356 @@
+
+  return {
+    token: "ce0fac19508f6c8f20066d345d360fd0",
+    "JS Analytics Version": 3,
+    distinct_id: identifier,
+    "Anonymized Builder Id": sha256(identifier),
+    "Anonymized Bundle Id": sha256(bundleId),
+    "Realm Version": realmVersion,
+    Binding: "Javascript",
+    Version: packageJson.version,
+    Language: language,
+    "Language Version": languageVersion,
+    Framework: framework,
+    "Framework Version": frameworkVersion,
+    "Host OS Type": os.platform(),
+    "Host OS Version": os.release(),
+    "Host CPU Arch": os.arch(),
+    "Node.js version": process.version,
+    "Core Version": realmCoreVersion,
+    "Sync Enabled": true,
+    "Installation Method": installationMethod[0],
+    "Installation Method Version": installationMethod[1],
+    "Runtime Engine": jsEngine,
+  };
+}
+
+/**
+ * Collect and send analytics data to MongoDB over HTTPS
+ * @param  {boolean} dryRun if true, collect data but don't submit
+ */
+async function submitAnalytics(dryRun) {
+  const data = await collectPlatformData();
+  const payload = {
+    webHook: {
+      event: "install",
+      properties: data,
+    },
+  };
+  doLog(`payload: ${JSON.stringify(payload)}`);
+
+  if (dryRun) {
+    doLog("Dry run; will not submit analytics");
+    return;
+  }
+
+  if (isAnalyticsDisabled()) {
+    doLog("Analytics is disabled");
+    return;
+  }
+
+  if ("REALM_PRINT_ANALYTICS" in process.env) {
+    console.log("REALM ANALYTICS", JSON.stringify(data, null, 2));
+  }
+
+  return new Promise((resolve, reject) => {
+    // node 19 turns on keep-alive by default and it will make the https.get() to hang
+    // https://github.com/realm/realm-js/issues/5136
+    const requestUrl = getAnalyticsRequestUrl(payload);
+
+    https
+      .get(requestUrl, { agent: new https.Agent({ keepAlive: false }) }, (res) => {
+        resolve({
+          statusCode: res.statusCode,
+          statusMessage: res.statusMessage,
+        });
+      })
+      .on("error", (error) => {
+        const message = error && error.message ? error.message : error;
+        const err = new Error(`Failed to dispatch analytics: ${message}`);
+        reject(err);
+      });
+  });
+}
+
+const optionDefinitions = [
+  {
+    name: "dryRun",
+    type: Boolean,
+    multiple: false,
+    defaultValue: false,
+    description: "If true, don't submit analytics",
+  },
+  { name: "log", type: Boolean, multiple: false, defaultValue: false, description: "If true, print log messages" },
+  { name: "test", type: Boolean, multiple: false, defaultValue: false, description: "If true, run as --dryRun --log" },
+];
+
+const options = commandLineArgs(optionDefinitions, { camelCase: true });
+
+let dryRun = options.dryRun;
+let log = options.log;
+
+if (options.test) {
+  dryRun = true;
+  log = true;
+}
+
+if (log) {
+  doLog = (msg) => console.log(msg);
+} else {
+  // eslint-disable-next-line no-unused-vars
+  doLog = () => {
+    /* don't log */
+  };
+}
```

Comment:
In general, I don't really like this pattern of reassigning a function - as something might capture it in a scoped variable and it won't update as expected.

In general, I would rather that we implement a (const) `log` function that does a check on the `options.log` before calling `console.log` or similar.

Or even better, that we just rely on the "debug" package, which we already have a dependency on - that way we could get rid of the `--log` runtime parameter, leaving just `--dry-run` which could be replaced with the `REALM_DISABLE_ANALYTICS` (which is already essentially doing the same anyway) - and if we did that we could drop the `command-line-args` dependency entirely. Just a thought ...

Reply:
Good point. I'll rewrite it to use the `debug` package and see if I can get rid of `command-line-args`